### PR TITLE
Ts migrate upgrade table contents

### PR DESCRIPTION
**Pull request by Mousticke**
Update component to typescript.

### CODE REVIEW

**Comment by minimalsm** [comment]
Thanks @Mousticke <3

File: src/components/UpgradeTableOfContents.tsx [hunk 1]
```diff
@@ -80,24 +95,40 @@ const TableOfContentsLink = ({ depth, item }) => {
...
 
-const ItemsList = ({ items, depth, maxDepth }) =>
-  depth <= maxDepth && !!items
-    ? items.map((item, index) => (
+const ItemsList = ({
+  items,
+  depth,
+  maxDepth,
+}: {
+  items: Item[]
```
Comment (pettinarip) [inline src/components/UpgradeTableOfContents.tsx]:
```suggestion
  items: Array<Item>
```

**Review by pettinarip — APPROVED** [review]
Thanks @Mousticke! I've taken the opportunity to refactor some types but your changes look great!

File: src/components/UpgradeTableOfContents.tsx (hunk 1 above)
Comment (Mousticke) [reply src/components/UpgradeTableOfContents.tsx]: Ok great! Thanks for the advice

**Pull request merged**